### PR TITLE
Add IoC and unit tests

### PR DESCRIPTION
**Pull request by Lcv004**

### CODE REVIEW

File: test/IoC/IoC.Test/IoCTest.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    [Fact]
+    public void GetService_ServiceNotInstantiated_ShouldNullException()
+    {
+        // Given
+        Ioc.GetInstance();
+        // When
+        
+        // Then
+        Assert.Throws<KeyNotFoundException>(() => Ioc.GetService("InexistentService"));
```
Comment (reynoldmorel) [inline test/IoC/IoC.Test/IoCTest.cs]: You meant `Unexistent`?

File: src/IoC/IoC.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+namespace IoC;
+
+public class Ioc
+{
+    private static Ioc? _instance;
+    private static Dictionary<string, object> _serviceInstances = new Dictionary<string, object>();
+    
+    private Ioc()
+    {
+        _serviceInstances.Add("ProductRepository", new ProductRepository());
```
Comment (reynoldmorel) [inline src/IoC/IoC.cs]: This is not meant to only register revices. Maybe use a more generic name like `_iocInstanceDict` or `_iocInstanceMap`

File: src/IoC/IoC.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        _serviceInstances.Add("ProductController", new ProductController(
+                            (ProductRepository)_serviceInstances["ProductRepository"]));
+    }
+
+    public static object GetService(string key)
```
Comment (reynoldmorel) [inline src/IoC/IoC.cs]: Let's just call this function `Get`

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
Please rename tests according to these changes

File: test/IoC/IoC.Test/IoCTest.cs (hunk 1 above)
Comment (Lcv004) [reply test/IoC/IoC.Test/IoCTest.cs]: Yes! It was a typo. Thank you.

File: src/IoC/IoC.cs (hunk 2 above)
Comment (Lcv004) [reply src/IoC/IoC.cs]: Understood!

File: src/IoC/IoC.cs (hunk 3 above)
Comment (Lcv004) [reply src/IoC/IoC.cs]: Done.

**Comment by Lcv004** [comment]
I already did changes to tests names accordingly.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**